webapp/graphite/render/datalib.py

```diff
@@ -158,6 +169,19 @@ def datapoints(self):
     timestamps = range(int(self.start), int(self.end) + 1, int(self.step * self.valuesPerPoint))
     return list(zip(self, timestamps))
 
+  @property
+  def tags(self):
+    return self.__tags
+
+  @tags.setter
+  def tags(self, tags):
+    if isinstance(tags, Tags):
+      self.__tags = tags
+    elif isinstance(tags, dict):
+      self.__tags = Tags(tags)
```

Comment:
TIL that this calls `__setitem__` on each item when type casting with `Tags(tags)`, didn't know that

Reply:
Yeah, I learned a lot about making custom dictionaries very quickly to throw this together